core/arch/aarchxx/mangle.c

```diff
@@ -3238,6 +3244,14 @@ mangle_exclusive_load(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
                 opnd_get_size(instr_get_src(in, 0)) ==
                     opnd_get_size(instr_get_dst(instr, 0)) &&
                 opnd_get_base(instr_get_dst(in, 0)) == base_reg &&
+                /* pick_scratch_reg() only takes 3 conflicts, so we push a pair with
```

Comment:
I'm thinking if we should use the un-optimized sequence for all load-shares-store-result cases. May be better to avoid adding more complexity.

Reply:
I considered this, but rejected it since this does not seem to be a rare occurrence: every single time there's a store-release pair, the load pair has one as XZR and the other as the store's destination.  I think the compiler really wants to use XZR for both, but making both load dests identical is undefined behavior, so it picks a dead register.  So it seems that this is not uncommon, and it's always in a single block, so it seems worth avoiding all the extra memory operations of the slowpath.

Reply:
Added a comment.

Reply:
Thanks.